### PR TITLE
Command line argument for setting/overriding Amazon S3 ACL

### PR DESCRIPTION
**Pull request by radek-senfeld**
Found a few moments so here it is.. ;-)

### CODE REVIEW

File: s3/s3.go [hunk 1]
```diff
@@ -190,6 +191,12 @@ const (
 	maxSizeForCopy = 5 * 1024 * 1024 * 1024 // The maximum size of object we can COPY
 )
 
+// Globals
+var (
+	// Flags
+	ACL = pflag.StringP("s3-acl", "", "", "Canned ACL used when creating buckets and/or storing objects in S3")
```
Comment (ncw) [inline s3/s3.go]: This should start with a lower case letter otherwise it is exported.  I'd call it s3ACL probably.

File: s3/s3.go [hunk 2]
```diff
@@ -346,7 +353,7 @@ func NewFs(name, root string) (fs.Fs, error) {
 		c:                  c,
 		bucket:             bucket,
 		ses:                ses,
-		acl:                fs.ConfigFile.MustValue(name, "acl"),
+		acl:                fs.ConfigFile.MustValue(name, "acl", *ACL),
```
Comment (ncw) [inline s3/s3.go]: This makes the config file override the command line argument - I think it should probably be the other way round.

**Comment by ncw** [comment]
Great!

I've written a few comments inline.

Also can you write something about the new flag in the s3 docs please?

Thanks

Nick

**Comment by radek-senfeld** [comment]
Like so, sir?

Not sure how to rebase / squash these commits.. Could you please help me?

**Comment by ncw** [comment]
Normally you would do it like this

    git checkout master
    git pull
    git checkout your-branch
    git rebase master

However you've used the master branch, so I think you'll need to do

    git pull --rebase

Then use this to squash the commits (you'll need to read up on exactly how it works)

    git rebase -i commit_number

Use `git log` to pick the commit number before you started making modifications.

You then need to push it with `--force` to the remote branch.

However I'm happy to do that if you don't feel confident doing it.

I was hoping for an update to `docs/content/s3.md` describing the new flag if you've got time?

**Comment by radek-senfeld** [comment]
Docs already updated. Not verbose enough? ;-)

I'll try to rebase once you confirm.. :-)

**Comment by ncw** [comment]
Something strange happened to your docs update - it looks like it got lost in a merge, so sorry I didn't see it.

I rescued the docs update from the diff.

This turned into rather a tricky merge so I have done it and committed it to master in bc414b698dd3acea3762a908f41612f451e36f68

Sorry for the delay, and thank you very much for your contribution :-)

**Comment by radek-senfeld** [comment]
Hi Nick,
I also fixed command-line overriding as you pointed before. This part seems to be missing.. Could you please merge it as well?

```
@@ -3,12 +3,15 @@
 		c:                  c,
 		bucket:             bucket,
 		ses:                ses,
-		acl:                fs.ConfigFile.MustValue(name, "acl", *s3ACL),
+		acl:                fs.ConfigFile.MustValue(name, "acl"),
 		root:               directory,
 		locationConstraint: fs.ConfigFile.MustValue(name, "location_constraint"),
 		sse:                fs.ConfigFile.MustValue(name, "server_side_encryption"),
 		storageClass:       fs.ConfigFile.MustValue(name, "storage_class"),
 	}
+	if *s3ACL != "" {
+		f.acl = *s3ACL
+	}
 	if *s3StorageClass != "" {
 		f.storageClass = *s3StorageClass
 	}
```

**Comment by ncw** [comment]
Thanks @radek-senfeld - I've merged that in cb40511807a03e922bd3fdd477fa41191bbfa62a - I managed to miss the notification when it came in so apologies for the delay!

**Comment by radek-senfeld** [comment]
Thank you @ncw!

**Pull request closed without merging**